### PR TITLE
Create 2020-09-02-solidity-0.7.1-release-announcement.md

### PR DESCRIPTION
**Pull request by franzihei**

### CODE REVIEW

File: _posts/2020-09-02-solidity-0.7.1-release-announcement.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+---
+
+[Solidity v0.7.1](https://github.com/ethereum/solidity/releases/tag/v0.7.1) .
+
+
+## Notable New Features 
+
+### Functions At File-Level
+
+Functions can now be defined at file-level. Such functions are also called "free functions"
```
Comment (ekpyron) [inline _posts/2020-09-02-solidity-0.7.1-release-announcement.md:17]:
```suggestion
Functions can now be defined at file-level. Such functions are called "free functions"
```

File: _posts/2020-09-02-solidity-0.7.1-release-announcement.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+(as opposed to functions bound to a specific contract).
+
+These functions will always be internal functions
+and are meant to replace internal library functions and their very special behaviour.
+
+These functions can do everything an internal library function can do, which includes
+calling other file-level functions, emitting events, sending Ether,
+calling functions on other contarcts "in the name" of the current contract
```
Comment (ekpyron) [inline _posts/2020-09-02-solidity-0.7.1-release-announcement.md:25]:
```suggestion
calling functions on other contracts "in the name" of the current contract
```

Comment (ekpyron) [reply _posts/2020-09-02-solidity-0.7.1-release-announcement.md:25]: Also I'm not sure I like ``"in the name" of the current contract``...
Maybe ``"in the name of the current contract" (i.e. as if the function was an internal function of the current contract)``?

Actually, maybe ``Similarly to internal library functions, free functions behave as if they were internal functions of the contract calling them, i.e. they can ... .`` or something like that might simplify the whole thing...

**Review by ekpyron — COMMENTED** [review]
Some comments

File: _posts/2020-09-02-solidity-0.7.1-release-announcement.md (hunk 1 above)
Comment (chriseth) [reply _posts/2020-09-02-solidity-0.7.1-release-announcement.md:17]: But we call them "file level function" all the time, too.

Comment (ekpyron) [reply _posts/2020-09-02-solidity-0.7.1-release-announcement.md:17]: Nowhere here, though, do we :-)? ``Functions At File-Level`` in the section header is not a name, but just says where the functions are...
If we *want* to call them "file level functions" all the time as well, then maybe this should just say ``Such functions are called "file level functions" or "free functions".`` :-). But yeah... maybe I'm nitpicking... I'd be fine with not changing this, it just seemed weird to me when I read it.

**Comment by chriseth** [comment]
Updated.

File: _posts/2020-09-02-solidity-0.7.1-release-announcement.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+### AST Export Directly After Parsing
+
+The Solidity compiler can now export the AST of a source file before import statements are
+applied. This feature allows to determine the dependency graph of a project without
+compiling everything.
+
+You can use it on the commandline via ``solc --stop-after parsing`` and though standard-json
+via ``{"settings": {"stopAfter": "parsing"}}``.
+
+Currently, it only this value is implemented for ``stopAfter``, but we plan to also add
```
Comment (ekpyron) [inline _posts/2020-09-02-solidity-0.7.1-release-announcement.md:78]:
```suggestion
Currently, it only this value is implemented for ``parsing``, but we plan to also add
```
right :-)?

**Review by ekpyron — COMMENTED** [review]
Reads good to me now - apart from the one comment and the still missing full changelog.

**Comment by chriseth** [comment]
I'll rebase and add the missing entry.

**Pull request merged**